src/com/facebook/buck/android/DummyRDotJava.java

```diff
@@ -119,9 +123,28 @@ public DummyRDotJava(
           androidResourceDeps,
           rDotJavaSrcFolder,
           forceFinalResourceIds,
-          unionPackage);
+          unionPackage,
+          /* rName */ Optional.<String>absent());
       steps.add(mergeStep);
-      javaSourceFilePaths = mergeStep.getRDotJavaFiles();
+
+      if (!finalRName.isPresent()) {
+        javaSourceFilePaths = mergeStep.getRDotJavaFiles();
+      } else {
+        MergeAndroidResourcesStep mergeFinalRStep = MergeAndroidResourcesStep.createStepForDummyRDotJava(
```

Comment:
Ugh, sorry.  This is over 100 chars, so it failed our lint check, so the automatic land failed.  Are you able to reformat?  If not, I can figure out how to import this to my laptop and do it manually.